mesonbuild/modules/gnome.py

```diff
@@ -1055,47 +1056,93 @@ def yelp(self, state: 'ModuleState', args: T.Tuple[str, T.List[str]], kwargs: 'Y
                 raise MesonException('Yelp requires a list of sources')
         elif args[1]:
             mlog.warning('"gnome.yelp" ignores positional sources arguments when the "sources" keyword argument is set')
-        source_str = '@@'.join(sources)
+        sources_files = [mesonlib.File.from_source_file(state.environment.source_dir,
+                                                        os.path.join(state.subdir, 'C'),
+                                                        s) for s in sources]
 
         langs = kwargs['languages']
+        if not langs:
+            langs = read_linguas(os.path.join(state.environment.source_dir, state.subdir))
+
+        media = kwargs['media']
+        symlinks = kwargs['symlink_media']
+        targets: T.List['Target'] = []
+        potargets: T.List['build.RunTarget'] = []
+
+        install_dir = os.path.join(state.environment.get_datadir(), 'help')
+        c_install_dir = os.path.join(install_dir, 'C', project_id)
+        c_data = build.Data(sources_files, c_install_dir, c_install_dir,
+                            mesonlib.FileMode(), state.subproject)
+        targets.append(c_data)
+
+        media_files = []
+        for m in media:
+            f = mesonlib.File.from_source_file(state.environment.source_dir,
+                                               os.path.join(state.subdir, 'C'), m)
+            media_files.append(f)
+            m_install_dir = os.path.join(c_install_dir, os.path.dirname(m))
+            m_data = build.Data([f], m_install_dir, m_install_dir,
+                                mesonlib.FileMode(), state.subproject)
+            targets.append(m_data)
+
+        pot_file = os.path.join('@SOURCE_ROOT@', state.subdir, 'C', project_id + '.pot')
+        pot_args = ['itstool', '-o', pot_file] + [os.path.join('@SOURCE_ROOT@', state.subdir, 'C', s) for s in sources]
```

Comment:
I think we should be using `state.find_program(`itstool')` here, not just a raw string. Among other things, that makes overriding the program in the machine file work

Reply:
Yes, fixed. Actually, this was originally the reason for me implementing this functionality :)

I have only used it for the `RunTarget`s. Is it also needed in the `command` arg or `CustomTarget`s? Or will the logic in `CustomTarget` take care of it?

Reply:
I got that that is the case in matrix. So I also added them for CustomTargets.